src/core/utils/crypto/pkcs12.test.ts

```diff
@@ -0,0 +1,141 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import {
+  CA_CERT_PATH,
+  ES_KEY_PATH,
+  ES_CERT_PATH,
+  ES_P12_PATH,
+  ES_P12_PASSWORD,
+  ES_EMPTYPASSWORD_P12_PATH,
+  ES_NOPASSWORD_P12_PATH,
+} from '@kbn/dev-utils';
+import { readFileSync } from 'fs';
+
+import { readPkcs12Keystore, Pkcs12ReadResult, readPkcs12Truststore } from './pkcs12';
```

Comment:
I added test cases for all of these except:

> 2. the .p12 certificate isn't base64 encoded

The .p12 file is always expected to be in binary format (the standard specifies BER, which is a binary form)., We use `readFileSync(path, 'base64')` to read in the binary and re-encode it as Base64, because that's what `node-forge` requires. So I don't think there's any need to add an extra test case here.